types/mithril/index.d.ts

```diff
@@ -208,7 +208,7 @@ declare namespace Mithril {
 	 * Any Javascript object that has a view method can be used as a Mithril component.
 	 * Components can be consumed via the m() utility.
 	 */
-	interface Component<Attrs, State extends Lifecycle<Attrs, State>> extends Lifecycle<Attrs, State> {
```

Comment:
Optimally, this would be `State extends this` (the state member prototypically inherits from the component), but since TypeScript doesn't support that, we settled for this constraint instead. So 👎  for this change.